concent_api/core/views.py

```diff
@@ -1275,13 +1287,21 @@ def create_file_transfer_token(
     """
     Function to create FileTransferToken from ReportComputedTask message
     """
-    current_time    = get_current_utc_timestamp()
     task_id         = report_computed_task.task_to_compute.compute_task_def['task_id']
     subtask_id      = report_computed_task.task_to_compute.compute_task_def['subtask_id']
     file_path       = '{}/{}/result'.format(subtask_id, task_id)
 
+    if operation == 'upload':
+        token_expiration_deadline = (
+            report_computed_task.task_to_compute.compute_task_def['deadline'] + 3 * settings.CONCENT_MESSAGING_TIME + 2 * settings.MAXIMUM_DOWNLOAD_TIME
+        )
+    else:  # It has to be 'download'
```

Comment:
Please add `if operation == 'download'` and assert before that if that `operation in ['upload', 'download']

Reply:
I would do sth like:
```
if operation == 'upload':
    # some code
elif if operation == 'download':
    # code
else:
 raise SomeException()
```

BTW: Is there a unit test for it?

Reply:
I agree with @rwrzesien. I think that using a [contract](https://en.wikipedia.org/wiki/Design_by_contract) to state which parameter values are supported is better than defensively checking for values that make no sense. By adding an `assert` at the beginning we state that any values other than `'upload'` or `'download'` are simply not supported and trying to call it with such a value is a bug in the calling code. The caller is responsible for ensuring that `operation` is one of these allowed values. It the caller gets it from the outside, he should validate it and report an error if the input is invalid. If he generates the values himself, then validation is not required. Only the caller can discern between these two cases.